### PR TITLE
Express checking in routes if that user is signed in before accessing that user's data

### PR DESCRIPTION
**Pull request by ami-domadia**

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -26,6 +26,15 @@ app.use(session({
 // authentication router
 app.use('/auth', require('./routes/user'));
 
+app.use((req, res, next) => {
+    console.log('I am in the middleware for checking logged in session');
+    if (req.params.userId && (req.session.userId !== req.params.userId)) {
```
Comment (chem2016) [inline server/index.js]: @ami-domadia , do you know how would be test this. For example, accessing a user's data?

**Review by chem2016 — COMMENTED** [review]
Could you tell us how did you test this implementation?

**Comment by ami-domadia** [comment]
Hi @chem2016 I really tried to make middleware for this to work but it didn't. So I now replaced it with logic in all the routes where userId is passed in. This seems to work.

You can test as follows:
1] Without logging in
http://localhost:3000/api/carts/user/0d0abbff-e8a1-4013-be63-e247eb8e8a2d-error 500  (test1@email.com)
http://localhost:3000/api/carts/user/0c3e02af-f388-4857-8ad5-23a7893d4bd5-error 500 (test2@email.com)

2] Login as test2@email.com
http://localhost:3000/api/carts/user/0d0abbff-e8a1-4013-be63-e247eb8e8a2d-error 500  (test1@email.com)
http://localhost:3000/api/carts/user/0c3e02af-f388-4857-8ad5-23a7893d4bd5-ok 200 (test2@email.com)
http://localhost:3000/api/carts/user/0d0abbff-e8a1-4013-be63-e247eb8e8a2d-error 500 (test1@email.com)

**Review by chem2016 — APPROVED** [review]
I tested this commits, user can only access its own cart.

**Pull request merged**